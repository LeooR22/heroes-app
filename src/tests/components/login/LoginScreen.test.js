import { mount } from "enzyme";
import { MemoryRouter, Route, Routes } from "react-router-dom";
import AuthContext from "../../../auth/authContext";
import LoginScreen from "../../../components/login/LoginScreen";
import types from "../../../types/types";

const mockNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockNavigate,
}));

describe("test en <LoginScreen/>", () => {
  const contextValue = {
    dispatch: jest.fn(),
    user: {
      logged: false,
    },
  };
  const wrapper = mount(
    <AuthContext.Provider value={contextValue}>
      <MemoryRouter initialEntries={["/login"]}>
        <Routes>
          <Route path="/login" element={<LoginScreen />} />
        </Routes>
      </MemoryRouter>
    </AuthContext.Provider>
  );
  test("debe de hacer match with snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de realizar el dispathch y la navegacion", () => {
    const handleClick = wrapper.find("button").prop("onClick");
    handleClick();

    //dispatch(...{name:"Fernado"})
    expect(contextValue.dispatch).toHaveBeenCalledWith({
      type: types.login,
      payload: {
        name: "Leonel",
      },
    });
    //mockNavigate = (/ , {replace: true})
    expect(mockNavigate).toHaveBeenCalledWith("/marvel", { replace: true });

    localStorage.setItem("lastPath", "/dc");

    handleClick();
    expect(mockNavigate).toHaveBeenCalledWith("/dc", {
      replace: true,
    });
  });
});
