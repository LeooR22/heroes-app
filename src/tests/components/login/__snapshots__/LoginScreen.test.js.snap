// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en <LoginScreen/> debe de hacer match with snapshot 1`] = `
<div
  className="container mt-5"
>
  <h1>
    Login Screen
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
